pkg/resources/cloudcontroller/hetzner.go

```diff
@@ -0,0 +1,139 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudcontroller
+
+import (
+	"fmt"
+
+	"k8c.io/kubermatic/v2/pkg/resources"
+	"k8c.io/kubermatic/v2/pkg/resources/reconciling"
+	"k8c.io/kubermatic/v2/pkg/resources/vpnsidecar"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	HetznerCCMDeploymentName = "hcloud-cloud-controller-manager"
+)
+
+var (
+	hetznerResourceRequirements = corev1.ResourceRequirements{
+		Requests: corev1.ResourceList{
+			corev1.ResourceMemory: resource.MustParse("50Mi"),
+			corev1.ResourceCPU:    resource.MustParse("100m"),
+		},
+	}
+)
+
+func hetznerDeploymentCreator(data *resources.TemplateData) reconciling.NamedDeploymentCreatorGetter {
+	return func() (string, reconciling.DeploymentCreator) {
+		return HetznerCCMDeploymentName, func(dep *appsv1.Deployment) (*appsv1.Deployment, error) {
+			dep.Labels = resources.BaseAppLabels(HetznerCCMDeploymentName, nil)
+
+			dep.Spec.Replicas = resources.Int32(1)
+
+			dep.Spec.Selector = &metav1.LabelSelector{
+				MatchLabels: resources.BaseAppLabels(HetznerCCMDeploymentName, nil),
+			}
+
+			podLabels, err := data.GetPodTemplateLabels(HetznerCCMDeploymentName, dep.Spec.Template.Spec.Volumes, nil)
+			if err != nil {
+				return nil, err
+			}
+
+			dep.Spec.Template.ObjectMeta = metav1.ObjectMeta{
+				Labels: podLabels,
+			}
+
+			dep.Spec.Template.Spec.DNSPolicy, dep.Spec.Template.Spec.DNSConfig, err =
+				resources.UserClusterDNSPolicyAndConfig(data)
+			if err != nil {
+				return nil, err
+			}
+
+			f := false
+			dep.Spec.Template.Spec.AutomountServiceAccountToken = &f
+
+			openvpnSidecar, err := vpnsidecar.OpenVPNSidecarContainer(data, "openvpn-client")
+			if err != nil {
+				return nil, fmt.Errorf("failed to get openvpn sidecar: %v", err)
+			}
+
+			credentials, err := resources.GetCredentials(data)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get credentials: %v", err)
+			}
+
+			network := data.Cluster().Spec.Cloud.Hetzner.Network
+			if network == "" {
+				network = data.DC().Spec.Hetzner.Network
+			}
+
+			dep.Spec.Template.Spec.Volumes = getVolumes()
+
+			dep.Spec.Template.Spec.Containers = []corev1.Container{
+				*openvpnSidecar,
+				{
+					Name:  HetznerCCMDeploymentName,
+					Image: data.ImageRegistry(resources.RegistryDocker) + "/hetznercloud/hcloud-cloud-controller-manager:v1.8.1",
+					Command: []string{
+						"/bin/hcloud-cloud-controller-manager",
+						"--kubeconfig=/etc/kubernetes/kubeconfig/kubeconfig",
+						"--cloud-provider=hcloud",
+						"--leader-elect=false",
+						"--allow-untagged-cloud",
+						"--allocate-node-cidrs=true",
+						"--cluster-cidr=10.244.0.0/16",
```

Comment:
This should usually equal to the pod subnet of the cluster, which is `172.25.0.0/16` if I'm not mistaken. Should we update this flag to reflect the correct value?

Reply:
I am not sure exactly. I can try, but leaving it on the default value from the hcloud-ccm did not seem to cause issues.

Reply:
Let's try to change it and see are tests going to pass.

Reply:
Tests seem to continue to work :)